figures/views.py

```diff
@@ -392,6 +393,125 @@ def get_serializer_context(self):
+        course_keys = figures.sites.get_course_keys_for_site(site)
+        date_for = datetime.utcnow().date()
+        month_for = '{}/{}'.format(date_for.month, date_for.year)
+        # TODO: NEEDS PAGINATION. Doing all initial to test endpoint
+        # TODO: Improvement: do 'courselike' to allow polymorphism
+        data = []
+        for course_key in course_keys:
+            data.append(metrics.get_month_course_metrics(site=site,
+                                                         course_id=str(course_key),
+                                                         month_for=month_for))
+        return Response(data)
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        TODO: Make sure we have a test to handle invalid or empty course id
+        """
+        site, course_id = self.site_course_helper(kwargs.get('pk', ''))
+        date_for = datetime.utcnow().date()
+        month_for = '{}/{}'.format(date_for.month, date_for.year)
+        data = metrics.get_month_course_metrics(site=site,
+                                                course_id=course_id,
+                                                month_for=month_for)
+        return Response(data)
+
+    @detail_route()
+    def active_users(self, request, **kwargs):
+        site, course_id = self.site_course_helper(kwargs.get('pk', ''))
+        date_for = datetime.utcnow().date()
+        months_back = 6
+        active_users = metrics.get_course_mau_history_metrics(
+            site=site,
+            course_id=course_id,
+            date_for=date_for,
+            months_back=months_back,
+        )
+        data = dict(active_users=active_users)
+        return Response(data)
+
+    @detail_route()
+    def course_enrollments(self, request, *args, **kwargs):
+        site, course_id = self.site_course_helper(kwargs.get('pk', ''))
+        data = dict(course_enrollments=self.historic_data(
+            request=request,
+            site=site,
+            course_id=course_id,
+            func=metrics.get_course_enrolled_users_for_time_period))
+        return Response(data)
+
+    @detail_route()
+    def num_learners_completed(self, request, *args, **kwargs):
+        site, course_id = self.site_course_helper(kwargs.get('pk', ''))
+        data = dict(num_learners_completed=self.historic_data(
+            request=request,
+            site=site,
+            course_id=course_id,
+            func=metrics.get_course_num_learners_completed_for_time_period))
+        return Response(data)
+
+    @detail_route()
+    def avg_days_to_complete(self, request, *args, **kwargs):
+        site, course_id = self.site_course_helper(kwargs.get('pk', ''))
+        data = dict(avg_days_to_complete=self.historic_data(
+            request=request,
+            site=site,
+            course_id=course_id,
+            func=metrics.get_course_average_days_to_complete_for_time_period))
+        return Response(data)
+
+    @detail_route()
```

Comment:
The `@detail_route` is going to be [deprecated in DRF==3.8 (Juniper)](https://www.django-rest-framework.org/community/3.8-announcement/#deprecations).

```suggestion
    @action(detail=True)
```

Not a deal breaker for now, but it's good to be aware of it. [Quoting the DRF docs](https://www.django-rest-framework.org/community/3.8-announcement/#deprecations):

> **Deprecations**
> action decorator replaces list_route and detail_route
> #5705 list_route and detail_route have been merge into a single action decorator. This improves viewset action introspection, and will allow extra actions to be displayed in the Browsable API in future versions.
> 
> Both list_route and detail_route are now pending deprecation. They will be deprecated in 3.9 and removed entirely in 3.10.
> 
> The new action decorator takes a boolean detail argument.
> 
> Replace detail_route uses with @action(detail=True).
> Replace list_route uses with @action(detail=False).

Reply:
@OmarIthawi I know. Problem is that the `action` decorator does not exist in DRF 3.6.3, so we're forced to use the list and detail decorators